packages/entities/entities-plugins/src/components/free-form/Datakit/flow-editor/store/init.ts

```diff
@@ -141,7 +153,9 @@ export function initEditorState(
   // Mark all nodes that should be in 'request' phase
   markRequestNodes(['request', 'service_request'])
 
-  return { nodes, edges, isEditing }
+  // TODO(Makito): Should we clean up stale nodes from the UI data in case the name of a deleted node is reused in the future?
```

Comment:
I don't think we should "remember" the UI config of a deleted node. Once it's deleted in the config, we should delete everything related.

Reply:
Agreed. Let's remove these dangling nodes from the UI data if they do not have corresponding ones in the config on init in the future.